Platon/backend/app/workspace_system/views.py

```diff
@@ -645,6 +668,209 @@ def delete(user_id, self):
+                    })
+            
+            # Pagination functionality
+            number_of_pages = 1
+            if form.page.data is not None and form.per_page.data is not None:
+                per_page = form.per_page.data
+                number_of_pages = math.ceil(len(followings_list) / per_page)
+                # Assign the page index to the maximum if it exceeds the max index
+                page = form.page.data if form.page.data < number_of_pages else number_of_pages-1
+                return_list = return_list[page*per_page:(page+1)*per_page]
+            return make_response(jsonify({'number_of_pages': number_of_pages ,'result': return_list}),200)
+            
+            
+        else:
+            return make_response(jsonify({'error': 'Input Format Error'}), 400)
+
+    @api.doc(responses={401 : 'Account Problems', 400 : 'Input Format Error' ,500 : ' Database Connection Error', 404: 'Not found'})
+    @api.expect(post_milestone_parser)
+    @login_required
+    @workspace_exists(param_loc='form',workspace_id_key='workspace_id')
+    @active_contribution_required(param_loc='form',workspace_id_key='workspace_id')
+    def post(user_id, self):
+        '''
+            Creates milestone
+        '''
+        form = PostMilestoneForm(request.form)
+        if form.validate():
+            
+            try: 
+                # Create milestone record
+                milestone = Milestone(user_id, form.workspace_id.data, form.title.data, form.description.data, form.deadline.data)
+            except:
+                return make_response(jsonify({'error': 'milestone record is not created'}), 500)
+
+            try:
+                db.session.add(milestone)
+                db.session.commit()
+            except:
+                return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+
+            return make_response(jsonify({'msg': 'milestone is successfully created'}), 200)
+
+        else:
+            return make_response(jsonify({'error': 'Input Format Error'}), 400)
+
+    @api.doc(responses={401 : 'Account Problems', 400 : 'Input Format Error' ,500 : ' Database Connection Error', 404: 'Not found'})
+    @api.expect(put_milestone_parser)
+    @login_required
+    @workspace_exists(param_loc='form',workspace_id_key='workspace_id')
+    @active_contribution_required(param_loc='form',workspace_id_key='workspace_id')
+    def put(user_id, self):
+        '''
+            Updates a milestone
+        '''
+        form = PutMilestoneForm(request.form)
+        if form.validate():
+            
+            try:
+                # Gets the parameters from the form data.
+                new_attributes = {}
+                for key, value in form.data.items():
+                    if value or (value == 0):
+                        # these attributes shouldn't be changed
+                        if key == 'milestone_id' or key == 'workspace_id':
```

Comment:
Instead of adding an if block here, you can modify your form in "forms.py" in a way only to get the parameters except the "milestone_id" and "workspace_id".

Reply:
However to change the requested milestone, I'll need milestone_id and workspace_id as parameters to make the query search. If I don't have the milestone_id for example, how can I know which milestone will be updated? 

Also workspace_id is for using decorators of active contribution and if workspace exists.